components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -98,7 +98,7 @@ class SystemEngineView @JvmOverloads constructor(
 
     override fun onDestroy() {
         session?.apply {
-            webView.destroy()
+            (webView.parent as? SystemEngineView)?.removeView(webView)
```

Comment:
Please add a comment to explain why we're doing this: specifically, the cyclical parent-child reference and that some apps hold long-lived references to this view, and thus the WebView, via the session manager

Reply:
Please also explain why `destroy` should not be called and where it gets called instead.

Reply:
OK, added comment.